test/test_nn.py

```diff
@@ -11351,6 +11351,14 @@ def test_adaptive_log_softmax(self):
 
         self.assertEqual(asfm(x, y).output.size(), (2, ))
 
+        # test no_batch_dim support
+        asfm = nn.AdaptiveLogSoftmaxWithLoss(16, 20, [5, 10, 15], div_value=2.)
+        x = torch.randn(1, 16)
+        y = torch.tensor([17])
+        x2 = x.squeeze(0)
+        y2 = y.squeeze(0)
+        self.assertEqual(asfm(x, y).output.squeeze(0), asfm(x2, y2).output)
```

Comment:
Thanks for adding the tests! Note that there is some precedent for using a reference function alongside the old style dictionary-based tests (see [here](https://github.com/pytorch/pytorch/blob/251686fc4cb1962944ed99c938df2d54f3d62e46/torch/testing/_internal/common_nn.py#L1315-L1329) for an example).

It won't work directly here because this module returns multiple outputs, but perhaps this can be made a bit more consistent with what's been done before